R/MsBackendDataFrame-functions.R

```diff
@@ -217,3 +217,28 @@ MsBackendDataFrame <- function() {
             list(numeric())
     res
 }
+
+#' @description
+#'
+#' Subset the `DataFrame` of a `MsBackendDataFrame` *by rows*.
+#'
+#' @param x `MsBackendDataFrame
+#'
+#' @param i `integer`, `character` or `logical`.
+#'
+#' @return Subsetted `x`
+#'
+#' @author Johannes Rainer
+#'
+#' @importMethodsFrom S4Vectors extractROWS
+#'
+#' @importFrom methods slot<-
+#'
+#' @noRd
+.subset_backend_data_frame <- function(x, i) {
+    if (missing(i))
+        return(x)
+    i <- i2index(i, length(x), rownames(x@spectraData))
+    slot(x, "spectraData", check = FALSE) <- extractROWS(x@spectraData, i)
```

Comment:
I guess slot(..., check = FALSE) <- is faster than @... because of the avoided checks?

Reply:
It avoids checking that the `value` is of the correct type. That's how it's done in `S4Vectors` that's why I've used it - but I haven't seen any major speed improvement.